src/rgw/rgw_rest_s3.cc

```diff
@@ -1705,11 +1705,13 @@ int RGWPostObj_ObjStore_S3::get_policy()
       err_msg = "Missing signature";
       return -EINVAL;
     }
-
     RGWUserInfo user_info;
 
     op_ret = rgw_get_user_info_by_access_key(store, s3_access_key, user_info);
     if (op_ret < 0) {
+      S3AuthFactory aplfact(store, s->account_name);
+      RGWLDAPTokenExtractor token_extr(s);
```

Comment:
I don't have the larger context of the code, but should we define an RGWLDAPTokenExtractor object if ldap was not configured to be used? maybe different scoping is possible?